R/layer.R

```diff
@@ -360,8 +360,8 @@ Layer <- ggproto("Layer", NULL,
   compute_statistic = function(self, data, layout) {
     if (empty(data))
       return(data_frame0())
-
-    self$computed_stat_params <- self$stat$setup_params(data, self$stat_params)
+    params <- defaults(self$stat_params, self$stat$default_params)
+    self$computed_stat_params <- self$stat$setup_params(data, params)
```

Comment:
3. Before we add computed components to the parameters, we initialise the parameters from the new field